### ISSUE TITLE
Redesign Directory using search/filter plugin

### ISSUE THREAD

**Issue by cjyabraham** [cncf/wordpress-site#57]
Redesign the [Speaker's Bureau directory](https://www.cncf.io/speakers/) to use the [search/filter plugin](https://www.cncf.io/projects/case-studies/) to fulfill the following requirements:
- make the profiles more compact so more can be displayed on one page; show only City, Country, Certification badges, and Areas of Expertise to the right of the picture and make every speaker box the same size
- Use the following fields for search filters: Certifications, Area of Expertise, Country, Self-Identification

Steps:
- [x] Diego: Design new directory page which works within the constraints of the search/filter plugin
- [x] Implement using the search/filter plugin

**Comment by cjyabraham** [comment]
@dankohn would like to imitate the geography feature we use for Events We'll Be At
(country and city nested).

**Comment by dankohn** [comment]
@deigolocco Can we please get a new design mockup showing a much denser layout?

**Comment by deigolocco** [comment]
sure @dankohn

**Comment by pablodelarocha** [comment]
@dankohn and @cjyabraham here's a design for this page according to your requests:
![Speakers-Bureau](https://user-images.githubusercontent.com/54405572/64559672-3ba9b200-d31d-11e9-877e-4c42d5865ae4.jpg)
(we can have more than just 9 per page... up to you on how many rolls should each page have)
Let us know your thoughts about it.

**Comment by KimMcMahon** [comment]
@pablodelarocha Can you update the design? 

We do not want the social profiles. Instead, if they have any certifications, we want that in the social profile place. 
They can up to 3 certifications: [CKA](https://www.cncf.io/certification/cka/), [CKAD](https://www.cncf.io/certification/ckad/), or [Ambassador](https://www.cncf.io/people/ambassadors/).

Please use the logo for each of the certifications. Can you do some mockups with all 3 certifications for some people and 1 and 2 certifications for others?

**Comment by pablodelarocha** [comment]
![image](https://user-images.githubusercontent.com/54405572/64569227-2ccdfa00-d333-11e9-9d19-7384ccb54373.png)
@KimMcMahon here's a suggestion of design as requested. Since the space to add the badges is very small and you can't read the badge's content, as you can see in the first tile, i've designed  tooltip that would appear when you hover the mouse over the badge, showing the name of the certificate.
Let us know what you think of that!

**Comment by dankohn** [comment]
Please also add a  ✈️ emoji if they're willing to travel and a  🖥 if they're willing to videoconference. And can we eliminate area of expertise and do first few lines of their bio instead. Area of expertise and language spoken seem better to get via the faceted search than have them displayed.

**Comment by dankohn** [comment]
If you make the emoji the same size as the badges, I think we're basically there.

**Comment by pablodelarocha** [comment]
@dankohn ok! Something like this?
![image](https://user-images.githubusercontent.com/54405572/64571531-8a197980-d33a-11e9-9f5a-ef78d6be0c5b.png)

**Comment by dankohn** [comment]
LGTM

**Comment by cjyabraham** [comment]
Looks good.  Can you show a mobile view?

**Comment by pablodelarocha** [comment]
@cjyabraham i'm sending attached a couple of options:
#1 simulates the exact same behavior as desktop, but instead of hovering with the mouse you have to touch the badge with your finger to read the tooltip
![Speakers-Bureau_mobile1](https://user-images.githubusercontent.com/54405572/64638545-568f2b80-d3dc-11e9-99a2-36587fe0393e.jpg)
or
#2 in case you don't find the tooltip very intuitive, we're suggesting this option with all badges with their description already open
![Speakers-Bureau_mobile2](https://user-images.githubusercontent.com/54405572/64638651-91915f00-d3dc-11e9-8389-b5299f7099fc.jpg)
Please, let us know if any of these suggestions works for you.

**Comment by dankohn** [comment]
I think #1 is OK. I'm not expecting most people to pick their speakers from their phone. Also, if we use SVG images, you can zoom in by pinching to see the logo better.

**Comment by deigolocco** [comment]
guys we are working in the frontend and during the integration with [search/filter plugin](https://www.cncf.io/projects/case-studies/) we realized the plugin doesn't work for users, but only for CPTs. 
@cjyabraham should we keep the Ultimate Member's filtering or implemente it manually as we did for webinars?

**Comment by cjyabraham** [comment]
Hmm.  Ok.  I'm fine with you implementing your custom filters then.

**Comment by deigolocco** [comment]
ok @cjyabraham , thanks.

**Comment by KimMcMahon** [comment]
@deigolocco Status? When will I have something to test?  Thanks!

**Comment by deigolocco** [comment]
@KimMcMahon we're planning to finish until next Monday.

This filtering has some challenges like:
- There's no taxonomies to filter, the filters will work with `user_metas`
- Some user_metas are serialized arrays, so this make the filtering harder and affect the performance
- we can not reuse the Ultimate Member's filters

We're working to have the filters like search/filter plugin. With counters, interacting with each other, ajax filtering and with acceptable performance.

**Comment by KimMcMahon** [comment]
@deigolocco Thank you.
Can we split this task into two? Update the profiles on the web page? 

Then, work on the filtering?

**Comment by deigolocco** [comment]
sure @KimMcMahon, we can finish the frontend tomorrow and continue the filtering in parallel

**Comment by deigolocco** [comment]
@KimMcMahon the profiles up to date
https://fz-57-cncf.pantheonsite.io/speakers/

**Comment by dankohn** [comment]
That shows 9 on a page. I think it should be more like 27.

In the dropdown menus, can you please make the lists single spaced.

**Comment by KimMcMahon** [comment]
@deigolocco Looks great!

Two things:
1) @deigolocco On the mobile version, can you add an arrow or something to move from page to page?

![File](https://user-images.githubusercontent.com/29307455/65273754-21bb5c80-dadf-11e9-8b3b-49ade959987d.jpg)

2) @dankohn When the page is narrow, the boxes do not line up with the exact same height. Do you want this fixed?

<img width="1920" alt="Screen Shot 2019-09-19 at 1 08 39 PM (2)" src="https://user-images.githubusercontent.com/29307455/65273811-3b5ca400-dadf-11e9-9749-b5bee344c094.png">

**Comment by deigolocco** [comment]
@dankohn is adjusted

@KimMcMahon 
1 - We don't have control for this pagination, it comes from Ultimate Member's plugin.
I did a few tests but the plugin hide the arrows navigation because it doesn't fit at screen
<img width="362" alt="Captura de Tela 2019-09-19 às 16 17 41" src="https://user-images.githubusercontent.com/5631313/65278792-390c4100-db03-11e9-8ffb-8df60c4060bb.png">
We can reduce the size, but not the number of itens. Reducing the size will became mobile unfriendly.
But for the new filtering we'll be in control, so the pagination will be as in the rest of the website.

2 - We normalized the heights and centralized
<img width="1221" alt="Captura de Tela 2019-09-19 às 17 33 56" src="https://user-images.githubusercontent.com/5631313/65279023-ae781180-db03-11e9-9252-47fa911592e8.png">

**Comment by deigolocco** [comment]
guys the filtering + paginations is fully working
https://fz-57-f-cncf.pantheonsite.io/speakers/

But we don't know which badge should be associated to certification KCSP

**Comment by cjyabraham** [comment]
A few comments:

1. can you show 27 per page instead of 9 (unless @dankohn has changed his mind on this)
2. when you open the filter drop-down, it looks different than the drop-downs on [webinars](https://fz-57-f-cncf.pantheonsite.io/webinars/). Can you make them consistent?
3. also on [webinars](https://fz-57-f-cncf.pantheonsite.io/webinars/) you use the filter placeholders "All companies", "All projects", etc.  Can we do the same pattern for speakers: "All certifications", "All areas of expertise"...?
4. when you choose ["Brazil"](https://fz-57-f-cncf.pantheonsite.io/speakers/?country=Brazil), the Certifications dropdown disappears.  I wonder whether it would work better to just grey-out the dropdown to indicate that there are no longer any filter options available?
5. is it possible to do a hierarchical list of countries, like we do on [case studies](https://fz-57-f-cncf.pantheonsite.io/projects/case-studies/).  Actually, I assume that the Speaker's Bureau doesn't have the data to support this.  Right?
6. Can you change "Speaker’s filtering:" to "Speakers Filtering:" to be consistent with how you do it on Webinars and Events?

**Comment by dankohn** [comment]
KCSP is for companies, not individuals, so doesn't apply. The badge should just be CKA, CKAD, both, or neither.

**Comment by cjyabraham** [comment]
As discussed in #cncf-wordpress-dev, please remove the "Willing to Present via Video Conference" icons from the profile cards.  This field is being removed since we can assume that pretty much everyone would be willing to do this.

Also "Willing to Travel" is being removed but "Travel Range" will remain in place.  Perhaps we can use the airplane icon when the person has "Travel Range" of Domestic or International but not Local.  So it would work like this:
- "Local": no airplane icon on profile card
- "Domestic": white airplane with tooltip "Willing to travel domestically"
- "International": blue airplane with tooltip "Willing to travel internationally"

@dankohn @KimMcMahon any thoughts?

**Comment by dankohn** [comment]
Maybe train for domestic and airplane for international?
--
Dan Kohn <dan@linuxfoundation.org> +1-415-233-1000
Executive Director, Cloud Native Computing Foundation cncf.io
<https://www.cncf.io/>
dankohn.com <https://www.dankohn.com/> or book on my calendar: dankohn.com/c
<https://www.dankohn.com/c>

**Comment by deigolocco** [comment]
implemented the train for domestic and airplane for international @dankohn 

adjusted @cjyabraham , except item 5 as we discussed on slack
https://fz-57-f-cncf.pantheonsite.io/speakers/

**Comment by cjyabraham** [comment]
Looking good.

1. Can we hide the self-identification filter for now until we have data for it.  It looks odd showing up but always greyed-out.
2. If you go to the page, pick "Cloud Platforms" and then pick "Ambassador", and then re-open the certifications drop-down, you'll correctly get this with all options in the drop-down:
<img width="639" alt="Screen Shot 2019-09-27 at 10 19 43 AM" src="https://user-images.githubusercontent.com/463160/65739697-7169db80-e110-11e9-8cbe-767a9abca979.png">
If you, however, directly go to [that link](https://fz-57-f-cncf.pantheonsite.io/speakers/?sb_certifications=Ambassador&cncf_expertise=Cloud+Platforms), it only has "Ambassador" in the drop-down.  
<img width="639" alt="Screen Shot 2019-09-27 at 10 21 11 AM" src="https://user-images.githubusercontent.com/463160/65739746-98281200-e110-11e9-9f11-e91ebadccbb9.png">

Can this be fixed so that it always shows all options?

**Comment by deigolocco** [comment]
ajusted @christinevblum

**Comment by cjyabraham** [comment]
@deigolocco 1 seems to be fixed but 2 is still a problem

**Comment by KimMcMahon** [comment]
@cjyabraham @deigolocco Can someone help me understand what is needed to get the new design deployed on https://www.cncf.io/speakers/?

**Comment by cjyabraham** [comment]
@KimMcMahon once we are done QA, we can deploy.  Does everything look ok to you?

**Comment by deigolocco** [comment]
@cjyabraham item 2 looks like fine for me. If you pick "Cloud Platforms" and then pick "Ambassador", then refresh the [page](https://fz-57-f-cncf.pantheonsite.io/speakers/?sb_certifications=Ambassador&cncf_expertise=Cloud+Platforms&_lc=sb_certifications), all options still there.

Please let me know what is not working for you.

**Comment by deigolocco** [comment]
![Captura de Tela 2019-09-29 às 10 50 34](https://user-images.githubusercontent.com/5631313/65833531-c2600800-e2a7-11e9-9d08-89363d0369ac.png)
![Captura de Tela 2019-09-29 às 10 50 46](https://user-images.githubusercontent.com/5631313/65833533-c55af880-e2a7-11e9-9959-b1b987dad6f3.png)

**Comment by KimMcMahon** [comment]
Can we deploy with the icons? Filtering still needs some work. 

For example, we don’t have the option from the mobile version to search by name.

**Comment by deigolocco** [comment]
@KimMcMahon we followed @cjyabraham [instructions](https://github.com/cncf/wordpress-site/issues/57#issue-484353429):
- Use the following fields for search filters: Certifications, Area of Expertise, Country, Self-Identification
Search by name is not included at the design [approved](https://github.com/cncf/wordpress-site/issues/57#issuecomment-530937795) by you
https://user-images.githubusercontent.com/54405572/64571531-8a197980-d33a-11e9-9f5a-ef78d6be0c5b.png

Self-Identification filter was hidden because we don't have data yet.
I believe this issue is completed, if new fields are required we should create a new issue.

**Comment by cjyabraham** [comment]
> @cjyabraham item 2 looks like fine for me. If you pick "Cloud Platforms" and then pick "Ambassador", then refresh the [page](https://fz-57-f-cncf.pantheonsite.io/speakers/?sb_certifications=Ambassador&cncf_expertise=Cloud+Platforms&_lc=sb_certifications), all options still there.
> 
> Please let me know what is not working for you.

@deigolocco what you say is true.  However, if you pick them in the opposite order, first pick "Ambassador" and then "Cloud Platforms", you'll see this:

<img width="512" alt="Screen Shot 2019-09-29 at 9 34 06 PM" src="https://user-images.githubusercontent.com/463160/65834021-016e7780-e301-11e9-98ac-2f49266282e6.png">

You can also just go to [this link](https://fz-57-f-cncf.pantheonsite.io/speakers/?sb_certifications=Ambassador&cncf_expertise=Cloud+Platforms&_lc=cncf_expertise).

**Comment by cjyabraham** [comment]
@KimMcMahon we're almost there with the filtering and then we will deploy all together.  It's not possible to release just the icons at this point.

If you're now wanting to search by name we need to consider how that can be integrated in to the design.  Can you open a new issue for that?

**Comment by KimMcMahon** [comment]
@cjyabraham @deigolocco Everything looks good to me. I was testing the filtering on my mobile and desktop and saw no issues.

Can we deploy?

For the other fields - I will open a new issue, as you both suggested.

Thanks!

**Comment by deigolocco** [comment]
@KimMcMahon I believe we're ready to deploy

@cjyabraham adjusted according our last call

**Comment by KimMcMahon** [comment]
Super - thanks! After @cjyabraham deploys, I'll do Q&A tomorrow.  Thanks!

**Comment by cjyabraham** [comment]
@deigolocco there's a render issue on mobile view.  Go [here](https://fz-57-f-cncf.pantheonsite.io/speakers/?sb_certifications=CKA&cncf_expertise=Cloud+Platforms&_lc=sb_certifications) and you'll see that the first speaker doesn't have any certification badges.  They have actually floated off to the side.  See:

<img width="813" alt="Screen Shot 2019-10-01 at 12 12 25 PM" src="https://user-images.githubusercontent.com/463160/65935890-db51f000-e444-11e9-9fa7-58d722ccb485.png">

It seems to only affect that one person.

I know @KimMcMahon is wanting to get this live so I've deployed the branch but will leave this issue open until this is fixed.

**Comment by cjyabraham** [comment]
Could you also make the airplane icon more in line with the other icons?  So move up a little bit:
<img width="280" alt="Screen Shot 2019-10-01 at 1 09 35 PM" src="https://user-images.githubusercontent.com/463160/65938284-c5e0c400-e44c-11e9-902f-d0470df7e818.png">

**Comment by deigolocco** [comment]
both adjusted @cjyabraham

**Comment by KimMcMahon** [comment]
Thank you!!!

@deigolocco The airplane is still a little lower than the other icon on mobile (iPhone X) and desktop - maybe it just needs to be deployed.

**Comment by deigolocco** [comment]
@KimMcMahon the last adjustments weren't deploy yet, you can check [here](https://fz-57-f-cncf.pantheonsite.io/speakers/) for now

**Comment by cjyabraham** [comment]
deployed
